openapi-v1.yaml

```diff
@@ -6036,3 +6125,128 @@ paths:
           schema:
             $ref: "#/definitions/Error"
+
+  #
+  # Groups: Swagger does not allow optional path parameters so we have to define two endpoints
+  # /groups/{namespace} for the root group
+  # /groups/{namespace}/{id} for other groups
+  # This will add additional methods for root group like listGroup and listRootGroup
+  # To make UI clients simpler, in the backend we will allow requests with empty id
+  # and reroute them to the root group
+  #
+
+  /groups/{namespace}/{id}:
+    parameters:
+      - name: namespace
+        type: string
+        in: path
+        required: true
+        description: The namespace of the group
+      - name: id
+        type: string
+        in: path
+        required: true
+        description: The id of the group
+    get:
+      description: Returns the contents, assets and subgroups, of the group
+      operationId: listGroup
+      tags:
+        - groups
+      responses:
+        200:
+          description: the group contents
+          schema:
+            $ref: '#/definitions/GroupListing'
+        default:
+          description: error response
+          schema:
+            $ref: "#/definitions/Error"
+    post:
+      description: Creates a new, empty child subgroup in the same namespace and of the same type as the parent. The id, file_type and namespace of the request are ignored
+      operationId: createGroup
+      parameters:
+        - in: body
+          name: body
+          schema:
+            $ref: '#/definitions/GroupCreate'
+      tags:
+        - groups
+      responses:
+        200:
+          description: group created successfully
+          schema:
+            $ref: '#/definitions/GroupListing'
+        default:
+          description: error response
+          schema:
+            $ref: "#/definitions/Error"
+    patch:
+      description: Updates group. Change attributes and add/remove assets
+      operationId: updateGroup
+      parameters:
+        - in: body
+          name: body
+          schema:
+            $ref: '#/definitions/GroupUpdate'
+      tags:
+        - groups
+      responses:
+        204:
+          description: assets changed successfully
+        default:
+          description: error response
+          schema:
+            $ref: "#/definitions/Error"
+    delete:
+      description: Deletes the groups and all the subgroups recursively. The assets are not deleted and do not move to any other group
+      operationId: deleteGroup
+      tags:
+        - groups
+      responses:
+        204:
+          description: group deleted successfully
+        default:
+          description: error response
+          schema:
+            $ref: "#/definitions/Error"
+
+  /groups/{namespace}:
+    parameters:
+      - name: namespace
+        type: string
+        in: path
+        required: true
+        description: The namespace of the group
+    get:
+      description: Returns the subgroups, of the root group
```

Comment:
This description does not make sense. Is this suppose to list groups in a namespace?

Reply:
No, just the subgroups of the root group, the equivalent of `/`. There in no endpoint to get all the groups of a namespace